### PR TITLE
custom events api

### PR DESCRIPTION
**Pull request by brettguenther**
Fixes MOB-27

### CODE REVIEW

File: src/api/events.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import Device from '../device';
+
+class Events {
+  static async sendEvent(eventData={}) {
+    const { type, metadata } = eventData;
+
+    const deviceId = Device.getId();
+    const userId = Storage.getItem(Storage.USER_ID);
+
+    return Http.request('POST','events', { type, metadata, deviceId, userId })
```
Comment (kochis) [inline src/api/events.js:12]: nit: spacing after `','` (really need an eslint file in here)

File: src/index.js [hunk 2]
```diff
@@ -307,6 +308,14 @@ class Radar {
       })
       .catch(handleError(callback));
   }
+
+  static sendEvent(eventData, callback=defaultCallback) {
+    Events.sendEvent(eventData)
+    .then((response) => {
+      callback(null, { event: response.event, status: STATUS.SUCCESS, }, response);
+    })
+    .catch(handleError(callback));
```
Comment (kochis) [inline src/index.js:317]: nit: indent

File: test/api/events.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    beforeEach(() => {
+        httpStub = sinon.stub(Http, 'request');
+    });
+
+    const customEventType = 'app_open';
+    const customEventMetadata = {'source':'organic'};
+    const customEventData = {customEventType, customEventMetadata}
+
+    afterEach(() => {
+    Http.request.restore();
```
Comment (kochis) [inline test/api/events.test.js:27]: indent

**Review by kochis — APPROVED** [review]

File: src/api/events.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import Device from '../device';
+
+class Events {
+  static async sendEvent(eventData={}) {
+    const { type, metadata } = eventData;
+
+    const deviceId = Device.getId();
+    const userId = Storage.getItem(Storage.USER_ID);
+
+    return Http.request('POST', 'events', { type, metadata, deviceId, userId })
```
Comment (kochis) [inline src/api/events.js:12]: Do we need to support `createdAt`? https://radar.com/documentation/api#send-a-custom-event

**Review by kochis — APPROVED** [review]

File: src/api/events.js (hunk 4 above)
Comment (brettguenther) [reply src/api/events.js:12]: yeah - this would be valuable if the sdk gets used server side. let me add before a release is cut.

**Review by kochis — APPROVED** [review]

**Pull request merged**